File: frontend/src/types/notifications.ts
export interface NotificationRead {
  created_at: string;
  dedup_key: string;
  id: string;
  payload: Record<string, unknown>;
  read_at: string | null;
  receiver_id: string;
  scenario: string;
  source_id: string;
  updated_at: string;
}

export interface NotificationListRead {
  items: NotificationRead[];
  page: number;
  page_size: number;
  total: number;
}

export interface NotificationListQuery {
  page?: number;
  pageSize?: number;
  unread?: boolean;
}

export interface NotificationUnreadCountRead {
  count: number;
}

export interface NotificationReadAllResult {
  read_count: number;
}

File: backend/app/schemas/notifications.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

NotificationDeliveryModeValue = Literal["real_time", "daily_digest"]


class NotificationRead(BaseModel):
    id: UUID
    receiver_id: UUID
    scenario: str
    source_id: str
    payload: dict[str, object]
    dedup_key: str
    delivery_mode: NotificationDeliveryModeValue
    digest_sent_at: datetime | None
    read_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationListRead(BaseModel):
    items: list[NotificationRead]
    page: int
    page_size: int
    total: int


class NotificationUnreadCountRead(BaseModel):
    count: int


class NotificationReadAllResult(BaseModel):
    read_count: int


class NotificationPreferenceRead(BaseModel):
    scenario: str
    label: str
    description: str
    direct_related: bool
    enabled: bool
    delivery_mode: NotificationDeliveryModeValue


class NotificationPreferenceListRead(BaseModel):
    items: list[NotificationPreferenceRead]


class NotificationPreferenceUpdateItem(BaseModel):
    scenario: str
    enabled: bool
    delivery_mode: NotificationDeliveryModeValue = "real_time"


class NotificationPreferenceUpdate(BaseModel):
    preferences: list[NotificationPreferenceUpdateItem]
